### ISSUE TITLE
Move to GitLab

### ISSUE THREAD

**Issue by StoneyJackson** [LibreFoodPantry/Community#62]
What needs to be done?

- [ ] Transfer each project to GitLab/LibreFoodPantry
- [ ] Finalize initial board structure and configure board/workflow labels
- [ ] Delete GitHub stuff from docs in LibreFoodPantry/ProjectTemplate
- [ ] Disable repositories (including issue trackers and boards) in GitHub

What else?

**Comment by cradkowski** [comment]
@StoneyJackson 
I've started importing the LFP projects to GitLab. One issue I've noticed is that you import them from GitHub to the GitLab account you are signed in to then transfer each project from your account to the group you want them in. This causes all issues created on a project that it can't link with a GitLab account to be owned by me. See the [Issues page](https://gitlab.com/groups/LibreFoodPantry/-/issues) for examples of this. Any ideas on what to do about this?

**Comment by cradkowski** [comment]
> What else?

- [ ] Enable DCO checking for each project.

**Comment by StoneyJackson** [comment]
Interesting... It looks like it still attributes the original poster/commenter, but yes you own it. Here is a screenshot of one of the issues from BEAR Necessities.

![Screen Shot 2019-08-08 at 2 24 48 PM](https://user-images.githubusercontent.com/1394670/62727940-93848e80-b9e8-11e9-80ec-d961664771e9.png)

So the question is, who should own issues and posts that don't have a corresponding account in GitLab? I'm thinking prior shop managers should import the projects they started. Then they will "own" the issues created during their watch. I'm assuming this process is straight forward. Does this make sense?

In the long run, this problem should resolve itself as new issues will be created by GitLab account holders.

@cradkowski Since you are an expert on this now, what documentation should we be following? I found some stuff, but I would rather not post the wrong stuff and confuse anyone.

**Comment by StoneyJackson** [comment]
> > What else?
> 
> * [ ]  Enable DCO checking for each project.

@cradkowski Perfect. Added.

**Comment by cradkowski** [comment]
@StoneyJackson 
>  I'm thinking prior shop managers should import the projects they started. Then they will "own" the issues created during their watch. I'm assuming this process is straight forward. Does this make sense?

I agree with this idea. So, I'll delete the current projects on GitLab and prior shop managers can each import their own from GitHub then?

> @cradkowski Since you are an expert on this now, what documentation should we be following?

I've been using [this document](https://docs.gitlab.com/ee/user/project/import/github.html)

The process is quick, the only part not covered on this guide is that you need to transfer the imported project from your account to the LFP group.

To do this on GitLab go to the imported repository and click `Settings` `General` and then click on the `Advanced` section and scroll down to:
![image](https://user-images.githubusercontent.com/47544822/62729784-a9944e00-b9ec-11e9-80b3-095da70ebea8.png)
And click `Transfer project`

**Comment by kwurst** [comment]
@cradkowski @StoneyJackson I don't see this problem on https://gitlab.com/LibreFoodPantry/FoodKeeper-API and https://gitlab.com/LibreFoodPantry/Theas-Pantry. It looks to me like the commits on these repositories all have the correct owners. Am I missing something?

**Comment by cradkowski** [comment]
@kwurst 
This issue seems to be specifically about the issue tracker and pull (merge) requests and not commits. See [this section](https://docs.gitlab.com/ee/user/project/import/github.html#how-it-works) of GitLab's import documentation for more information about how it links accounts when importing.

Also, if you click on some of those accounts listed in the commits on GitLab, not all of them are appear to be real GitLab accounts. It seems like GitLab copies over the username from GitHub if they can't link it to an account when importing.

Specific Examples: https://gitlab.com/LibreFoodPantry/Theas-Pantry/commit/4e564b952b95f3d14e1d795ac0079285a7924fc3

And: https://gitlab.com/LibreFoodPantry/FoodKeeper-API/commit/4e658b990247a4f5f505196bab61deab506d323d
